### PR TITLE
api: stream: allow name update

### PR DESCRIPTION
**Pull request by gioelecerati**
**What does this pull request do? Explain your changes. (required)**

- Allow `name` as field in `PATCH` stream api

**Specific updates (required)**

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

**Screenshots (optional)**

**Checklist**

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by gioelecerati** [comment]
@victorges do you know why the stream name was immutable till now?
There was also a specific test making sure it was immutable (`the stream name is immutable`) - is a name update going to affect anything?

**Comment by victorges** [comment]
@gioelecerati oh wow super weird that there's even a test. I have no idea 😅 can't think of anything that would break but now it's better to test it anyway. Maybe on the very beginning it was used for configuring streams on mist (or even wowza lol)

**Comment by gioelecerati** [comment]
> @gioelecerati oh wow super weird that there's even a test. I have no idea 😅 can't think of anything that would break but now it's better to test it anyway. Maybe on the very beginning it was used for configuring streams on mist (or even wowza lol)

Coold :D was added here https://github.com/livepeer/studio/pull/430 and I think it was just added without any specific reason, but let's test throughly either way!

File: packages/api/src/controllers/stream.test.ts [hunk 1]
```diff
@@ -1240,6 +1240,13 @@ describe("controllers/stream", () => {
         });
       });
 
+      it("should allow patch of name", async () => {
+        const res = await client.patch(patchPath, {
+          name: "new name",
+        });
+        expect(res.status).toBe(204);
```
Comment (victorges) [inline packages/api/src/controllers/stream.test.ts:1247]: You should check if the database object is actually updated

**Review by victorges — APPROVED** [review]
neat

**Comment by victorges** [comment]
@gioelecerati lol got it. that PR has nothing to do with stream name then, probably just a confusing string i used to test the `additionalProperties` validation.

**Pull request merged**